modules/db2/src/main/java/org/testcontainers/containers/Db2Container.java

```diff
@@ -0,0 +1,129 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.WaitingConsumer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.LicenseAcceptance;
+import org.testcontainers.utility.LogUtils;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+
+public class Db2Container<SELF extends Db2Container<SELF>> extends JdbcDatabaseContainer<SELF> {
```

Comment:
the self-typing is not needed here, please use `extends JdbcDatabaseContainer<Db2Container>`

Reply:
can you elaborate on why self-typing is not needed? Won't removing self-typing mean that users cannot extend Db2Container with additional "wither" methods?

Reply:
we consider this a rare case, while the self typing here will make it hard to use with Kotlin/Scala

There are a few concrete container classes where the self typing is used, but mostly for the historical reasons, and we plan to drop the self typing in 2.x, see:
https://speakerdeck.com/bsideup/geecon-2019-testcontainers-a-year-in-review?slide=74